c++/src/kj/async.c++

```diff
@@ -831,24 +830,26 @@ struct FiberStack::Impl {
                         stackSize, PROT_READ | PROT_WRITE));
 
     // Stick `Impl` at the top of the stack.
-    Impl& impl = *(reinterpret_cast<Impl*>(reinterpret_cast<byte*>(stack) + allocSize) - 1);
+    Impl* impl = (reinterpret_cast<Impl*>(reinterpret_cast<byte*>(stack) + allocSize) - 1);
 
     // Note: mmap() allocates zero'd pages so we don't have to memset() anything here.
 
-    KJ_SYSCALL(getcontext(&impl.fiberContext.context));
-    impl.fiberContext.context.uc_stack.ss_size = allocSize - sizeof(Impl);
-    impl.fiberContext.context.uc_stack.ss_sp = reinterpret_cast<char*>(stack);
-    impl.fiberContext.context.uc_stack.ss_flags = 0;
-     // to reuse fibers the start fibers runs forever, so this doesn't
-     // need to point to the main context
-    impl.fiberContext.context.uc_link = 0;
+    KJ_SYSCALL(getcontext(context));
+    context->uc_stack.ss_size = allocSize - sizeof(Impl);
```

Comment:
Hmm, I think it would be clearer to move this initialization of `context` to happen inside `FiberStack`'s constructor, since it's no longer a member of `Impl`.

And then you don't have to change `impl` to be a pointer...

Reply:
It'll require doing the pointer math for the stack allocation again inside the `FiberStack` ctor as well as `Impl::alloc`.

Reply:
Done

Reply:
You are right, the way you did it made more sense. Sorry about that. I guess you can remove the last commit.

Can you rebase the PR to make the commit history cleaner? I think it would make sense to squash down to two commits:

1. Implement freelist of fiber stacks.
2. Switch to setjmp/longjmp.

I think this should be just a matter of `git rebase -i master` and then squash the first sequence of commits and then the second sequence, and update the resulting commit descriptions -- probably no need to reorder anything.